### ISSUE TITLE
`Physics.FindPairs` reports extraneous pairs when there is only one cell in the collision layer

### ISSUE THREAD

**Issue by slowhei** [Dreaming381/Latios-Framework#53]
Hello Latios community,

I found a bug in Psyshock on Latios Framework version 0.10.6.

On line 276 of "FindPairsInternal.cs", `Physics.FindPairs` tries to detect pairs between cells in the collision layer and the cross bucket:

```
var crossBucket = layer.GetBucketSlices(IndexStrategies.CrossBucketIndex(layer.cellCount));
    for (int i = 0; i < IndexStrategies.SingleLayerPart2Count(layer.cellCount); i++)
    {
        var bucket  = layer.GetBucketSlices(i);
        var context = new FindPairsBucketContext(in layer,
                                                 in layer,
                                                 in bucket,
                                                 in crossBucket,
                                                 jobIndex,
                                                 isThreadSafe,
                                                 isThreadSafe,
                                                 !isThreadSafe);
        if (processor.BeginBucket(in context))
        {
            FindPairsSweepMethods.BipartiteSweepCellCross(in layer,
                                                          in layer,
                                                          in bucket,
                                                          in crossBucket,
                                                          jobIndex,
                                                          ref processor,
                                                          isThreadSafe,
                                                          isThreadSafe,
                                                          !isThreadSafe);
            processor.EndBucket(in context);
        }
    jobIndex++;
}
```

However, if there is only one cell in the collision layer, these lines of code will treat that one cell as the cross bucket also and essentially sweep that one cell against itself. This results in extraneous pairs being reported by `Physics.FindPairs`, such as pairs where both A and B are the same entity.

### Steps to Reproduce

Build a collision layer with only one cell:

```
var settings = new CollisionLayerSettings {
    worldAabb = new Aabb(-300f, 300f),
    worldSubdivisionsPerAxis = new int3(1, 1, 1)
};
```

Use the overload of `Physics.BuildCollisionLayers` that accepts a collider bodies array and an AABB's array:
(Other overloads might work also, but I have not tried them.)

```
state.Dependency = Physics.BuildCollisionLayer(bodies: ..., overrideAbbs: ...)
    .WithSettings(settings)
    .ScheduleParallel(out CollisionLayer collisionLayer, Allocator.Persistent, state.Dependency)
```

In the subscene, make sure there are two colliders that do not overlap.

Run `Physics.FindPairs` and log pairs detected:

```
state.Dependency = Physics.FindPairs(
    collisionLayer,
    new Processor { }
).ScheduleParallel(state.Dependency);

struct Processor : IFindPairsProcessor {
     void IFindPairsProcessor.Execute(in FindPairsResult result) {
        Debug.Log($"Pair: {result.entityA.entity.Index}, {result.entityB.entity.Index}");
    }
}
```

You will see in the logs that there are pairs that consist of the same entity.

### Ideas for Solution

Adding `if (layer.cellCount <= 1) return;` before line 276 of "FindPairsInternal.cs" solves the issue if `Physics.FindPairs` runs immediately.

I am not sure what to do if `Physics.FindPairs` schedules parallel jobs and runs lines 119-226 of "FindPairsInternal.cs". I am not inclined to analyze it.

<hr/>

Please let me know if you need any more information or something doesn't add up.
Also, thanks for creating Psyshock! It has been a great learning resource for me.

**Comment by Dreaming381** [comment]
Great find!

Your fix solves the issue entirely for single-layer FindPairs, because all single-cell jobs will end up using that method. A similar fix is needed for line 827 in the same file.

**Comment by Dreaming381** [comment]
Fixed in 0.10.7
